src/js/oauth.js

```diff
@@ -1,90 +1,92 @@
-        ',width=' + w +
-        ',height=' + h +
-        ',top=' + top +
-        ',left=' + left);
+        var left = ((width / 2) - (w / 2)) + dualScreenLeft;
+        var top = ((height / 2) - (h / 2)) + dualScreenTop;
+        var newWindow = window.open(url, '_blank', 'location=0,menubar=0,status=0,toolbar=0' +
+            ',width=' + w +
+            ',height=' + h +
+            ',top=' + top +
+            ',left=' + left);
 
-    // Puts focus on the newWindow
-    if (window.focus) {
-        newWindow.focus();
+        // Puts focus on the newWindow
+        if (window.focus) {
+            newWindow.focus();
+        }
     }
-}
-
-function openImplicitOAuthWindow (scopes) {
-    var clientId = 'fa54866255ea641235e596e5659fa726a4aa9f7ecc72758f';
-    var redirectURI = 'https://dev.beam.pro/oauthreturn.html';
-    // var redirectURI = 'http://localhost:3000/oauthreturn.html';
-    var url = 'https://beam.pro/oauth/authorize?response_type=token&' +
-        'redirect_uri=' + redirectURI + '&' +
-        'scope=' + scopes + '&' +
-        'client_id=' + clientId;
-    popupCenter(url);
-}
-
-/**
- * Retrieve an implicit access token from the url hash
- */
-function getAccessToken () {
-    var hash = window.location.hash.slice(1);
 
-    if (hash.length === 0) {
-        return null;
+    function openImplicitOAuthWindow (scopes) {
+        var clientId = 'fa54866255ea641235e596e5659fa726a4aa9f7ecc72758f';
+        var redirectURI = 'https://dev.beam.pro/oauthreturn.html';
+        // var redirectURI = 'http://localhost:3000/oauthreturn.html';
+        var url = 'https://beam.pro/oauth/authorize?response_type=token&' +
+            'redirect_uri=' + redirectURI + '&' +
+            'scope=' + scopes + '&' +
+            'client_id=' + clientId;
+        popupCenter(url);
     }
 
-    var hashParts = hash.split('&');
-    for (let i = 0; i < hashParts.length; i++) {
-        var item = hashParts[i].split('=');
-        if (item[0] === 'access_token') {
-            return item[1];
+    /**
+     * Retrieve an implicit access token from the url hash
+     */
+    function getAccessToken () {
+        var hash = window.location.hash.slice(1);
+
+        if (hash.length === 0) {
+            return null;
         }
-    }
 
-    return null;
-}
+        var hashParts = hash.split('&');
+        for (let i = 0; i < hashParts.length; i++) {
+            var item = hashParts[i].split('=');
+            if (item[0] === 'access_token') {
+                return item[1];
+            }
+        }
 
-function handleOAuthResponse () {
-    var token = getAccessToken();
+        return null;
+    }
 
-    window.opener.replaceToken(token);
-    window.close();
-}
+    function handleOAuthResponse () {
+        var token = getAccessToken();
 
-function replaceToken (token) {
-    if (!token) {
-        $('.auth-token').html('Error retrieving token');
-        return;
+        window.opener.replaceToken(token);
```

Comment:
Does this still work? replaceToken probably needs to be explicitly attached to the window now